### ISSUE TITLE
Amplifications location in report

### ISSUE THREAD

**Issue by sbihel** [STAMP-project/dspot#362]
During the amplification process, only the number of a-amplifications and i-amplifications is logged. 

Do you think it is possible to also note the location of each amplification? 

The location could be the line number, of the path in the AST. But I am not sure if it is possible because multiple rounds of amplifications can occur and we would then need to be able to keep the locations up-to-date.

**Comment by danglotb** [comment]
Yes, it is possible to track the position of I-amplification / A-amplification.

In **DSpot**, there this notion of test parent, _i.e._ the test on which is amplified to obtain new amplified test. The top parent, is the manual - original test. This relationship can be represented by a tree.

If we walk this tree, we can have all the amplifications applied.

I agree that would be great to have such information.

We started to this by adding some comment on the amplified tests. The problem is that the amplified test became unreadable, and so, comments are, by default, disable. 

If we put this information into the json, which is possible I think we can completely remove comments.

**Comment by sbihel** [comment]
Ok I will work on this

**Comment by danglotb** [comment]
Okay, Great!

Do not hesitate to ask questions / suggest some stuff. I'll be glad to discuss! 

I'm gonna change this issue into a track of your progress on this feature.

Cheers.

**Comment by danglotb** [comment]
Hi @sbihel 

I think you won't gonna work on this anymore, right? I took the liberty to make it available.

**Comment by sbihel** [comment]
Hi @danglotb,

Yes sorry, I should have cleaned my mess 🙇‍♂️

**Comment by danglotb** [comment]
Hi @sbihel 

It is totally fine! I'll just unassign you from this issue to highlight the fact that this is should be done.

I hope you are doing great, and I'll see around!
